src/components/Profile/UserInfoForm.tsx

```diff
@@ -1,58 +1,82 @@
 import { FormEvent, useCallback, useState } from 'react';
 import styled from 'styled-components';
 
-import { BodyMedium } from 'components/core/Text/Text';
+import { BodyMedium, Caption } from 'components/core/Text/Text';
 import BigInput from 'components/core/BigInput/BigInput';
 import TextArea from 'components/core/TextArea/TextArea';
 import Spacer from 'components/core/Spacer/Spacer';
-import { USERNAME_REGEX } from 'utils/regex';
+import colors from 'components/core/colors';
 
 type Props = {
   className?: string;
-  onSubmit?: (event: FormEvent) => void;
-  handleIsValidChange: (isValid: boolean) => void;
-  mode?: string;
+  mode?: 'Add' | 'Edit';
+  onSubmit: () => void;
+  username: string;
+  onUsernameChange: (username: string) => void;
+  bio: string;
+  onBioChange: (bio: string) => void;
 };
 
+export const BIO_MAX_CHAR_COUNT = 500;
+
 function UserInfoForm({
   className,
   onSubmit,
-  handleIsValidChange,
+  username,
+  onUsernameChange,
+  bio,
+  onBioChange,
```

Comment:
these have to be passed in rather than set on the component, since higher-level parent may want to handle submission / error handling differently